app/src/main/java/com/tari/android/wallet/infrastructure/backup/WalletBackup.kt

```diff
@@ -0,0 +1,77 @@
+package com.tari.android.wallet.infrastructure.backup
+
+import com.tari.android.wallet.infrastructure.backup.compress.CompressionMethod
+import com.tari.android.wallet.infrastructure.security.encryption.SymmetricEncryptionAlgorithm
+import net.lingala.zip4j.ZipFile
+import net.lingala.zip4j.model.ZipParameters
+import net.lingala.zip4j.model.enums.AesKeyStrength
+import net.lingala.zip4j.model.enums.EncryptionMethod
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+
+
+interface WalletBackup {
+    fun run(key: CharArray): File
+
+    companion object {
+        fun defaultStrategy(cd: File): WalletBackup = CompressionOnlyBackup.defaultStrategy(cd)
+    }
+}
+
+val BACKUP_FILES = arrayOf("tari_wallet_db.sqlite3")
```

Comment:
Not sure how this one is used but it might be safer to have the database file name in `Constants.Wallet`. There we already have `Constants.Wallet.walletDBName`, maybe have a second constant for the full name with extension?